src/Client.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <thread>
+#include <zmq.hpp>
+
+#include "libursa/Json.h"
+#include "libursa/ZHelpers.h"
+#include "spdlog/spdlog.h"
+
+void status_worker(std::string server_addr, std::string conn_id) {
+    while (true) {
+        zmq::context_t context(1);
+        zmq::socket_t socket(context, ZMQ_REQ);
+        socket.setsockopt(ZMQ_LINGER, 0);
+        socket.setsockopt(ZMQ_RCVTIMEO, 1000);
+        socket.connect(server_addr);
+
+        s_send(socket, "status;");
+        std::string res_str;
+        unsigned int retries = 0;
+
+        while (res_str.empty()) {
+            if (retries == 30) {
+                spdlog::warn(
+                    "UrsaDB server seems to be unresponsive. Failed to obtain "
+                    "progress for more than 30 seconds.");
+            }
+
+            res_str = s_recv(socket);
+            retries++;
+        }
+
+        auto res = json::parse(res_str);
+        auto res_tasks = res["result"]["tasks"];
+
+        for (json::iterator it = res_tasks.begin(); it != res_tasks.end();
+             ++it) {
+            if ((*it)["connection_id"] == conn_id) {
+                unsigned int work_done = (*it)["work_done"].get<unsigned int>();
```

Comment:
Is the .get() necessary? Json is usually smart, so

```cpp
unsigned int work_done = (*it)["work_done"];
```

should work as well. The same applies below.